### PR TITLE
CTDA-1281 Add metrics collection to controllers and connectors

### PR DESCRIPTION
**Pull request by DavidGregory084**

### CODE REVIEW

File: app/metrics/Monitors.scala [hunk 1]
```diff
@@ -16,59 +16,13 @@
...
-trait RequestMonitor[A] {
-  val name: String
-
-  val path: String         = s"$name.request"
-  val timer: Timer         = Timer(path)
-  val callCounter: Counter = Counter(path)
-
-  def completionCounter(result: A): Option[Counter] = None
-  val failureCounter: Counter                       = Counter(s"$path.failed")
-}
```
Comment (DavidGregory084) [inline app/metrics/Monitors.scala:36]: A Dropwizard Metrics `Timer` is already a `Counter` of requests and a `Meter` of the rate of requests, so I have removed this request monitor abstraction. 

Also, the `grafana-dashboards` project expects metric names to follow a certain format - `<metric-name>` for the `DownstreamResponseTimesGraphBuilder` and `<metric-name>-success-counter` and `<metric-name>-failed-counter` in order to work with the `DownstreamResponsesPerMinuteGraphBuilder`

File: app/metrics/Monitors.scala [hunk 2]
```diff
@@ -16,59 +16,13 @@
...
-      case s if s <= 1000 => Counter("arrivals-per-eori-501-1000")
-      case s if s <= 2000 => Counter("arrivals-per-eori-1001-2000")
-      case s if s <= 3000 => Counter("arrivals-per-eori-2001-3000")
-      case _              => Counter("arrivals-per-eori-3001-or-more")
-    }
 
-  def countMessages(messageType: MessageType, channel: ChannelType, outcome: SubmissionProcessingResult): Counter =
-    Counter(s"message-received.${channel.toString}.${messageType.code}-${outcome.toString}")
+  def messageReceived(registry: MetricRegistry, messageType: MessageType, channel: ChannelType, outcome: SubmissionProcessingResult): Unit =
+    registry.meter(s"message-received.${channel.toString}.${messageType.code}.${SubmissionProcessingResult.format(outcome)}").mark()
```
Comment (DavidGregory084) [inline app/metrics/Monitors.scala:27]: This enables us to see count, m1, m5, m15 rates of message types and outcomes received

File: app/models/SubmissionProcessingResult.scala [hunk 3]
```diff
@@ -20,6 +20,17 @@ sealed trait SubmissionProcessingResult
...
 
+  def format(result: SubmissionProcessingResult) = result match {
+    case SubmissionSuccess =>
+      "success"
+    case SubmissionFailureInternal =>
+      "failure-internal"
+    case SubmissionFailureExternal =>
+      "failure-external"
+    case SubmissionFailureRejected(responseBody) =>
+      "rejected"
```
Comment (DavidGregory084) [inline app/models/SubmissionProcessingResult.scala:31]: I think previously the entire responseBody was `toString`ed into the metric name so I added this

**Review by amaalali — APPROVED** [review]
Looks good

**Pull request merged**